tests/functional/regressions/test_issue140.py

```diff
@@ -36,9 +37,21 @@ def proc(a):
     mm_A.register_obj_processors({"C1": proc})
     mm_B.register_obj_processors({"C1": proc})
 
-    scoping.MetaModelProvider.clear()
-    scoping.MetaModelProvider.add_metamodel("*.a", mm_A)
-    scoping.MetaModelProvider.add_metamodel("*.b", mm_B)
+    a_dsl = LanguageDesc(
+        name='a-dsl',
+        pattern='*.a',
+        description='Test Lang A',
+        metamodel=mm_A)
+
+    b_dsl = LanguageDesc(
+        name='b-dsl',
+        pattern='*.b',
+        description='Test Lang B',
+        metamodel=mm_B)
+
+    clear_language_registrations()
+    register_language(a_dsl)
+    register_language(b_dsl)
```

Comment:
I'm adding a way to call `register_language` with direct parameters. This allows for shorter form:
```python
register_language('a-dsl', '*.a', metamodel=mm_A)
```

Reply:
Next step, after rebasing....

Reply:
Done.

I have a question concerning rebasing:
I did ``` git rebase --onto origin/feature/language-registration ```: is this the way to go? I got a merge (which I did not understand completely)...?

Reply:
Since you have branched from `feature/language-registration` you don't have to use `--onto`. `--onto` is used only if you are rebasing on some other branch. See `man git-rebase` for some nice examples.

All commits from your branch are applied on the top of the base branch one by one. It's like you just did those changes/commits manually. Some of them might be in conflict with the changes in the base branch in which case you should resolve the conflicts (see `git status` and search for conflict markers in files). Once resolved you continue the rebase with `git rebase --continue`). At any time  you can abort the rebase and return to the previous state with `git rebase --abort`). If you run rebase in interactive mode you get a chance to squash, reword, split, augment etc. each commit by giving a special commands in the editor. That is a nice tool when you are need to tidy up the branch.

Tip: git keeps track of all branch positions (branches are like pointers) so if you understand this mechanism you can dissect and change your branch history with confidence as all commits you have made are there until garbage collecting kicks in when all dangling commits (without pointers -- refs: branches, tags pointing to them) are removed. To see this log of where your branch pointer was in the past see `git reflog`. And `git reset` can be used to move your branch pointer to any point in history. Of course, these operations should be used rarely and never on the published branch (e.g. never on `master`) as the published history should be immutable.

Reply:
Strange enough. I just investigated the branch history and it looks like you have do a regular merge of `feature/language-registration` to your branch. Never mind ;)